text/0000-build-std/5-stage-1b.md

```diff
@@ -0,0 +1,986 @@
+```
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why limit enabling standard library features to nightly?*][rationale-features]
+
+*See the following sections for future possibilities:*
+
+- [*Allow enabling/disabling features with build-std*][future-features]
+
+## Public and private dependencies
+[public-and-private-dependencies]: #public-and-private-dependencies
+
+Implicit dependencies on the standard library default to being public
+dependencies ([?][rationale-implicit-public]). When a standard library is
+explicitly written, then it will be private by default, like any other written
+dependency, unless explicitly marked as public ([?][rationale-explicit-private]).
+
+```toml
+[package]
+name = "hello_world"
+version = "0.1.0"
+edition = "2024"
+
+[dependencies]
+```
+
+..is equivalent to the following explicit dependency on `std`:
+
+```toml
+[package]
+name = "hello_world"
+version = "0.1.0"
+edition = "2024"
+
+[dependencies]
+std = { builtin = true, public = true }
+```
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why default to public for the implicit standard library dependencies?*][rationale-implicit-public]
+- [*Why follow the default privacy of explicit standard library dependencies?*][rationale-explicit-private]
+
+## `dev-dependencies` and `build-dependencies`
+[dev-dependencies-and-build-dependencies]: #dev-dependencies-and-build-dependencies
+
+Explicit dependencies on the standard library are not supported in
+`build-dependencies` ([?][rationale-no-deps-in-build-deps]).
+
+Implicit and explicit dependencies on the standard library are supported for
+`dev-dependencies` in the same way as regular `dependencies`.
+
+## Registries
+[registries]: #registries
+
+Standard library dependencies will be present in the registry index
+([?][rationale-cargo-index]). A `builtin_deps` key is added to the
+[index's JSON schema][cargo-json-schema] ([?][rationale-cargo-builtindeps]).
+`builtin_deps` is similar to the existing `deps` key and contains a list of JSON
+objects, each representing a dependency that is "builtin" to the Rust toolchain
+and cannot otherwise be found in the registry.
+
+> [!NOTE]
+>
+> It is expected that the keys of these objects will be:
+>
+> - `name`
+>   - String containing name of the `builtin` package. Can shadow the names of
+>     other packages in the registry (except those packages in the `deps` key
+>     of the current package) ([?][rationale-cargo-index-shadowing])
+>
+> - `features`:
+>   - An array of strings containing enabled features in order to support changing
+>     the standard library features on nightly. Optional, empty by default.
+>
+> - `optional`, `default_features`, `target`, `kind`:
+>   - These keys have the same definition as in the `deps` key.
+>
+> The keys `req`, `registry` and `package` from `deps` are not required per the
+> limitations on builtin dependencies.
+>
+> The key is optional and its default value will be the implicit builtin
```

Comment:
I'm assuming "The key" is `builtin_deps` since other keys are being talked about here